static/app/utils/duration/getDuration.tsx

```diff
@@ -44,26 +53,30 @@ const DURATION_LABELS = {
   ms: t('ms'),
   millisecond: t('millisecond'),
   milliseconds: t('milliseconds'),
+  us: 'μs', // SI units don't need a translation
+  microsecond: t('microsecond'),
+  microseconds: t('microseconds'),
 };
 
 export default function getDuration(
   seconds: number,
   fixedDigits: number = 0,
   abbreviation: boolean = false,
   extraShort: boolean = false,
-  absolute: boolean = false
+  absolute: boolean = false,
+  minimumUnit: number = MILLISECOND
```

Comment:
My only suggestion is that it might be helpful to add a comment explaining how `minimumUnit` works, but it is at least more intuitive than `extraShort` and `abbreviation` which are also unexplained 🤷

Reply:
Fair! I added JSDoc for the function while I was at it
<img width="378" alt="Screenshot 2025-01-10 at 4 12 30 PM" src="https://github.com/user-attachments/assets/8d2ed420-4111-4e3e-a3aa-8113c212602c" />